### ISSUE TITLE
Unused function reported as "Unused assignment"

### ISSUE THREAD

**Issue by hauserx** [cameron-martin/bazel-lsp#42]
Possibly that's issue with implementation in starlark-rust, but filling for now here before investigating further:

![image](https://github.com/cameron-martin/bazel-lsp/assets/32371993/3c96f938-5d5c-4b09-b6b0-fdc975526a55)

**Comment by cameron-martin** [comment]
Yeah it looks like starlark-rust [treats defs as assignments](https://github.com/facebook/starlark-rust/blob/781a4643e82f012487f06085f8c5209bcfb493fe/starlark/src/analysis/names.rs#L513). I suppose they kind of are, but it doesn't make for a very good error message.

**Comment by hauserx** [comment]
For variables also `unused assignment` may not be the best message:

![image](https://github.com/cameron-martin/bazel-lsp/assets/32371993/ff2ca19a-4848-495d-ab4c-f7d62363faf1)

I think it should be "Unused variable" for vars - https://github.com/bazelbuild/starlark/blob/master/spec.md#assignments

For functions it could be "Unused variable" (as Following spec https://github.com/bazelbuild/starlark/blob/master/spec.md#function-definitions ), but probably "Unused function" should be acceptable as  `def` always declare a function.

**Comment by hauserx** [comment]
Adding yet another message that possibly can be improved (`Used ignored variable`). 
In this case variable _a is not defined at all, so message that using some ignore value is confusing.

![image](https://github.com/cameron-martin/bazel-lsp/assets/32371993/9b08bc9b-b7c1-4ef8-a108-bcee664390a7)
